### PR TITLE
Added new wallet pkg with KeyManager and Account helpers from the iota-core mock package

### PR DESCRIPTION
**Pull request by alexsporn**
In this package we can implement common wallet operations

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,6 +16,7 @@ require (
 	github.com/pasztorpisti/qs v0.0.0-20171216220353-8d6c33ee906c
 	github.com/samber/lo v1.38.1
 	github.com/stretchr/testify v1.8.4
+	github.com/wollac/iota-crypto-demo v0.0.0-20221117162917-b10619eccb98
```
Comment (muXxer) [inline go.mod:19]: maybe we should use the fork?
should we rename the package in the fork so we don't need a replace statement?

File: wallet/account.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+}
+
+// ID returns the accountID.
+func (e *Ed25519Account) ID() iotago.AccountID {
+	return e.accountID
+}
+
+// Address returns the account address.
+func (e *Ed25519Account) Address() iotago.Address {
```
Comment (muXxer) [inline wallet/account.go]: That confused me a bit.
It's not the account address derived from the accountID, but the Ed25519 address derived from the private key. :sweat_smile: 
Can we change the function name and/or the comment?

File: wallet/account.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	return iotago.Ed25519AddressFromPubKey(ed25519PubKey)
+}
+
+// PrivateKey returns the account private key for signing.
+func (e *Ed25519Account) PrivateKey() ed25519.PrivateKey {
+	return e.privateKey
+}
+
+func AccountFromParams(accountHex string, privateKey string) Account {
```
Comment (muXxer) [inline wallet/account.go]: Maybe return the errors instead of panic?

File: wallet/account.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// Account represents an account.
+type Account interface {
+	// ID returns the accountID.
+	ID() iotago.AccountID
+
+	// Address returns the account address.
+	Address() iotago.Address
```
Comment (muXxer) [inline wallet/account.go]: See my comment in the implementation.

File: wallet/keymanager.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/wollac/iota-crypto-demo/pkg/bip32path"
+	"github.com/wollac/iota-crypto-demo/pkg/slip10"
+	"github.com/wollac/iota-crypto-demo/pkg/slip10/eddsa"
+
+	"github.com/iotaledger/hive.go/ierrors"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+const (
+	pathString = "44'/4218'/0'/%d'"
```
Comment (muXxer) [inline wallet/keymanager.go]: Maybe this should be a parameter to the keymanager instead.

**Review by muXxer — CHANGES_REQUESTED** [review]

File: wallet/account.go (hunk 2 above)
Comment (alexsporn) [reply wallet/account.go]: That is why I told you to wait with the review, exactly because I was refactoring this, hehe

File: wallet/account.go [hunk 6]
```diff
@@ -56,16 +62,16 @@ func (e *Ed25519Account) PrivateKey() ed25519.PrivateKey {
 	return e.privateKey
 }
 
-func AccountFromParams(accountHex string, privateKey string) Account {
+func AccountFromParams(accountHex string, privateKey string) (Account, error) {
 	accountID, err := iotago.AccountIDFromHexString(accountHex)
 	if err != nil {
-		panic(fmt.Sprintln("invalid accountID hex string", err))
+		return nil, ierrors.Wrapf(err, "invalid accountID hex string")
```
Comment (muXxer) [inline wallet/account.go]: Wrap

File: wallet/account.go [hunk 7]
```diff
@@ -56,16 +62,16 @@ func (e *Ed25519Account) PrivateKey() ed25519.PrivateKey {
...
 	}
 
 	privKey, err := crypto.ParseEd25519PrivateKeyFromString(privateKey)
 	if err != nil {
-		panic(fmt.Sprintln("invalid ed25519 private key string", err))
+		return nil, ierrors.Wrapf(err, "invalid ed25519 private key string")
```
Comment (muXxer) [inline wallet/account.go]: Wrap

File: wallet/account.go [hunk 8]
```diff
@@ -41,8 +44,11 @@ func (e *Ed25519Account) ID() iotago.AccountID {
 	return e.accountID
 }
 
-// Address returns the account address.
-func (e *Ed25519Account) Address() iotago.Address {
+func (e *Ed25519Account) Address() *iotago.AccountAddress {
+	return e.accountID.ToAddress().(*iotago.AccountAddress)
```
Comment (muXxer) [inline wallet/account.go:49]: nolint :D

File: go.mod (hunk 1 above)
Comment (alexsporn) [reply go.mod:19]: Lets do that later. We need to decide where to move the code to

**Review by muXxer — APPROVED** [review]

**Pull request merged**